app/questionnaire/rules.py

```diff
@@ -67,14 +67,42 @@ def evaluate_condition(condition, answer_value, match_value):
     comparison_operators = {
         'equals': lambda answer_value, match_value: answer_value == match_value,
         'not equals': lambda answer_value, match_value: answer_value != match_value,
-        'contains': lambda answer_value, match_value: isinstance(answer_value, list) and match_value in answer_value,
-        'not contains': lambda answer_value, match_value: isinstance(answer_value, list) and match_value not in answer_value,
+        'contains':
+            lambda answer_value, match_value: isinstance(answer_value, list)
+            and match_value in answer_value,
+        'contains all':
+            lambda answer_value, match_value: isinstance(answer_value, list)
+            and isinstance(match_value, list)
```

Comment:
Should types be checked in here? This seems like a check that  should be done within validator.

Reply:
They were being checked in here in the existing implementation so I mostly just decided to extend that, also feels like checking them here is simpler that in validator?